File: android/ValleCOM/app/src/main/java/com/valleapp/vallecom/adaptadores/AdaptadorComanda.java
package com.valleapp.vallecom.adaptadores;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

/**
 * Created by valle on 12/09/14.
 */
public class AdaptadorComanda extends FragmentPagerAdapter {

    Fragment comanda = null;
    Fragment teclados = null;

    public AdaptadorComanda(FragmentManager fm, Fragment comanda, Fragment teclados) {
        super(fm); this.comanda = comanda; this.teclados= teclados;
    }

    @Override
    public Fragment getItem(int i) {
        switch (i){
            case 1:
                 return comanda;
              case 0:
                 return  teclados;
             default:
                return null;

        }

    }

    @Override
    public int getCount() {
        return 2;
    }

    // Returns the page title for the top indicator
    @Override
    public CharSequence getPageTitle(int position) {
        switch (position){
            case 0:
                return  "Teclados";
            case 1:
                return "La comanda";
            default:
                return null;

        }

    }
}

File: android/ValleTPV/ValleCOM/src/main/java/com/valleapp/vallecom/adaptadores/AdaptadorComanda.kt
package com.valleapp.vallecom.adaptadores

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class AdaptadorComanda(
    fa: FragmentActivity,
    private val comanda: Fragment,
    private val teclados: Fragment
) : FragmentStateAdapter(fa) {

    private val tabTitles = listOf("Teclados", "Comandas")

    override fun getItemCount(): Int {
        return tabTitles.size // Debería ser 2 si siempre son "Mesas" y "Pedidos"
    }


    override fun createFragment(position: Int): Fragment {
        return when (position) {
            1 -> comanda
            0 -> teclados
            else -> throw IllegalArgumentException("Invalid position")
        }
    }

    fun getPageTitle(position: Int): CharSequence {
        return tabTitles[position]
    }
}
